### ISSUE TITLE
Upload progress indicator

### ISSUE THREAD

**Issue by abyrd** [conveyal/analysis-ui#282]
I'm uploading a 117MB zip of GTFS and OSM, and I've seen a spinner for a long time. I'm not sure if it's doing anything. Is there any way to show a progress bar on this operation?

**Comment by abyrd** [comment]
I think what happened is that the GTFS processing thread died, so the upload is reported "in progress" forever. The server side has no way to report an error. Created #283

**Comment by mattwigway** [comment]
The new analysis UI does not accept zipped bundles of GTFS and OSM; rather, if you want to upload multiple GTFS files you simply select all of them.

**Comment by mattwigway** [comment]
When a feed is processed successfully, the status is shown. I'll investigate why the error messages aren't working (#283).

**Comment by abyrd** [comment]
@mattwigway yes, I eventually realized this, but left these tickets open because the real subject is the display of error messages. I also created a ticket in the UI project for making it clearer that you could multiple-select GTFS feeds on upload.

**Comment by abyrd** [comment]
I'm reopening this ticket, because it is about adding an upload progress indicator. Although I now know why the upload failed, I still think it would be good in the long term to have a progress indicator.

**Comment by mattwigway** [comment]
I implemented this before, which was why I closed it, but it seems to not be in the current codebase. I'll figure out what happened to it this afternoon, apologies.

**Comment by mattwigway** [comment]
This was fixed in the `misc-fixes` branch but never merged. Looks like it has some conflicts, which @trevorgerhardt and I will resolve.

**Comment by abyrd** [comment]
I'm wondering if this was ever merged in. We should check whether it was and whether the functionality is available. Another place we need a progress indicator is uploading shapefiles, which can take several minutes. There's the upload itself, followed by rasterizing the shapefile attributes into a grid, followed by uploading all the grids to S3. Each of these provides log feedback so could conceivably report accurate percentage of progress to the UI.

**Comment by trevorgerhardt** [comment]
It was not merged in. `misc-fixes` was an out of date branch that contained a wide variety of changes that were mostly in conflict with the development branch by the time I deleted it. 

I agree that progress indicators are important.

**Comment by abyrd** [comment]
I'm bumping this issue just because today I'm uploading an OSM.PBF file that's "only" 82MB and find myself wondering what's going on. I click a button to create the region but from a user point of view nothing happens at all in the UI. If I click the button again it seems to maybe repeatedly upload the file. I suspect our users are clicking this button N times and continuously reuploading the same file.

**Comment by abyrd** [comment]
I'm noticing this again today uploading a relatively small GTFS. The UI is unconventional and probably very confusing to most users in that clicking the button to upload GTFS displays "loading..." for a long period of time, with no indication of whether the upload is progressing or stalled or the application has locked up. Also, the word "loading" is also not commonly used for uploads. We should change it to something like "uploading, please wait..."

**Comment by abyrd** [comment]
We definitely still want to add progress indicators, this came up at our recent roadmap meeting.
